diff --git a/CMakeLists.txt b/CMakeLists.txt
index 11f48c9..892a0bc 100644
--- a/monero-cpp/CMakeLists.txt
+++ b/monero-cpp_new/CMakeLists.txt
@@ -21,6 +21,7 @@ set(BUILD_LIBRARY ON)
 set(BUILD_SAMPLE OFF)
 set(BUILD_SCRATCHPAD OFF)
 set(BUILD_TESTS OFF)
+option(BUILD_STATIC "Build monero-cpp as a static library" OFF)
 
 ###################
 # monero-project
@@ -169,7 +170,7 @@ else()
   message(STATUS "Found libunbound include (unbound.h) in ${UNBOUND_INCLUDE_DIR}")
   if(UNBOUND_LIBRARIES)
     message(STATUS "Found libunbound library")
-    if (WIN32)
+    if (WIN32 OR NOT BUILD_STATIC)
       add_library(unbound STATIC IMPORTED)
     else()
       add_library(unbound SHARED IMPORTED)
@@ -280,7 +281,13 @@ set(
 )
 
 if (BUILD_LIBRARY)
-  add_library(monero-cpp SHARED ${LIBRARY_SRC_FILES})
+#  add_library(monero-cpp SHARED ${LIBRARY_SRC_FILES})
+
+  if (BUILD_STATIC)
+    add_library(monero-cpp STATIC ${LIBRARY_SRC_FILES})
+  else()
+    add_library(monero-cpp SHARED ${LIBRARY_SRC_FILES})
+  endif()
 
   target_include_directories(monero-cpp PUBLIC
       ${CMAKE_CURRENT_SOURCE_DIR}/include
